test/e2e-component-proxy/component_proxy.go

```
@@ -0,0 +1,121 @@
+			fresh.Spec.Proxy = operatorv1.AuthenticationProxyConfig{}
+		}
+		if _, err := clients.OperatorClient.OperatorV1().Authentications().Update(ctx, fresh, metav1.UpdateOptions{}); err != nil {
+			g.GinkgoWriter.Printf("cleanup: failed to restore proxy: %v\n", err)
+			return
+		}
+
+		g.GinkgoWriter.Println("cleanup: waiting for ProxyConfigControllerDegraded to clear")
+		if err := wait.PollUntilContextTimeout(ctx, 10*time.Second, 10*time.Minute, true, func(ctx context.Context) (bool, error) {
+			config, err := clients.OperatorClient.OperatorV1().Authentications().Get(ctx, "cluster", metav1.GetOptions{})
+			if err != nil {
+				return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 63) but it returns nil

(nilerr)